bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/http/RateLimitedHttpClient.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.http.internal.http;
+
+import java.net.URI;
+import java.util.concurrent.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.AuthenticationStore;
+import org.eclipse.jetty.client.api.Request;
+
+/**
+ * The {@link RateLimitedHttpClient} is a wrapper for a Jetty HTTP client that limits the number of requests by delaying
+ * the request creation
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class RateLimitedHttpClient {
+    private static final int MAX_QUEUE_SIZE = 1000; // maximum queue size
+    private HttpClient httpClient;
+    private int delay = 0; // in ms
+    private final ScheduledExecutorService scheduler;
+    private final LinkedBlockingQueue<RequestQueueEntry> requestQueue = new LinkedBlockingQueue<>();
+
+    private @Nullable ScheduledFuture<?> processJob;
+
+    public RateLimitedHttpClient(HttpClient httpClient, ScheduledExecutorService scheduler) {
+        this.httpClient = httpClient;
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * Stop processing the queue and clear it
+     */
+    public void shutdown() {
+        stopProcessJob();
+        requestQueue.forEach(queueEntry -> queueEntry.future.completeExceptionally(new InterruptedException()));
```

Comment:
Completing with an InterruptedException would be misleading, so I suggest just cancelling the futures instead. It will complete with a CancelationException instead.
```suggestion
        requestQueue.forEach(queueEntry -> queueEntry.future.cancel(false));
```